### DISCUSSION TITLE
How does barman-cloud-backup back up a postgres base backup?

### DISCUSSION THREAD

**Discussion by shauryagoel** [EnterpriseDB/barman#927] [Q&A] [answered]
I have installed Postgres on K8s and use `barman-cloud-backup` for weekly backup of the whole Postgres cluster to a Google Bucket. My PGDATA is around 22GB and the node on which Postgres runs has 8GB storage available. The backup is failing and I am getting the error- `The node was low on resource: ephemeral-storage. Container Postgres was using 16888Ki, which exceeds its request of 0`. Why is this error occurring?
Can someone explain to me how the `barman-cloud-backup` works? Does it store the data in a temporary storage of node and from there push the data to the bucket? Or Does it extract the data from PGDATA and upload that to the bucket? Or some other way? I currently have no compression enabled.

**Comment by martinmarques** [comment, marked as answer]
There was a bug in the barman-cloud implementation, which was fixed in Barman 3.13. Does this still happen now?
